lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+require 'rest-client'
+require 'timeout'
+require 'tempfile'
+require 'zip'
+require 'base64'
+
+module Terraform
+  class Runner
+    class << self
+      def available?
+        return @available if defined?(@available)
```

Comment:
ruby has a thing called double bang which can eliminate the need to check the `defined?` here.  If `@available` is `nil` or `false`, `!!@available` will be `false`.  If it's any other object, it will be `true`.

It uses negation twice to turn anything into a `true/false`. value and is best in `?` methods like this.   See: https://rubyquicktips.com/post/583755021/convert-anything-to-boolean

Reply:
In retrospect, this is fine as is.  Ignoring.